include/fmt/ranges.h

```diff
@@ -607,6 +612,47 @@ struct formatter<
   }
 };
 
+#ifdef __cpp_lib_ranges
```

Comment:
Making this formatter conditional on ranges can be problematic.  {fmt} already has replacement for `std::ranges::begin/end` (`detail::range_begin/end`) so let's use them here.

Reply:
`fmt::basic_string_view` would first need a constructor that accepts an iterator/sentinel pair (or iterator/iterator pair).

Reply:
string_view requires a pointer so iterator/sentinel should be converted to pointer/size which better be done outside of the ctor.

Reply:
Thinking more of it, I think it's OK to keep this conditional but let's at least kill the dependency on`<ranges>` and check `__cpp_lib_concepts` for contiguous iterator support instead.

Reply:
Reduced the scope of <ranges>.